app/src/main/java/com/bignerdranch/android/androidacademy/FragmentMoviesList.kt

```diff
@@ -22,11 +25,31 @@ class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
         listener = null
     }
 
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        view.findViewById<ViewGroup>(R.id.movie_list_item)
+
+        val rvMovie = view.findViewById<RecyclerView>(R.id.rv_movie)
+
+       // rvMovie.layoutManager = GridLayoutManager( context, 2)
+        val myAdapter = MovieAdapter(this, MoviesDataSource().getMovies())
+        rvMovie.adapter = myAdapter
+
+        view.findViewById<ViewGroup>(R.id.rv_movie)
                 .setOnClickListener {
                     listener?.onItemClickShowDetail()
                 }
+
+    }
+
+    override fun onItemClick(movie: Movie) {
+        listener!!.onItemClickShowDetail()
```

Comment:
here will be a NULL safe use like so `listener?. onItemClickShowDetail()`

Reply:
got it